### ISSUE TITLE
[P0] Label/Training tab — let users label their own patterns for training data

### ISSUE THREAD

**Issue by sauravpanda** [sauravpanda/swingflow#142]
## Why

Pattern identification is still wrong and we need ground truth to improve it. Instead of users manually writing JSON (see #139), give them a UI to label their own clips. High UX win (they can fix the AI's mistakes on their own videos) + generates training data at zero friction.

## UX

New top-level nav entry **"Label"** alongside Dashboard / Rhythm / Analyze / Practice.

### Flow

1. /label lists all the user's past analyses
2. Click an analysis → label editor page
3. Editor renders:
   - The same video player + pattern-timeline from the analysis view
   - AI-predicted patterns pre-populated as **editable** blocks (so the user is CORRECTING, not starting blank)
   - "Add label at $currentTime" button → dialog:
     - **Pattern family** dropdown (canonical WSDC list from prompts.py — sugar push, whip, etc.)
     - **Variant** dropdown (contextual to family)
     - **Count** (6 / 8 / custom)
     - **Start / end time** (pre-filled from playhead + 3s default, draggable)
     - **Confidence** slider (optional)
     - **Notes** (optional)
   - Each label block on the timeline shows:
     - Color: user-label (blue) vs AI-prediction (current colors) vs accepted-AI (green check)
     - Click to edit / delete
     - Keyboard shortcuts: J/K to prev/next block, D to delete, A to add, S to save
4. Top-right: "Export JSON" (training schema from #139) + "My labels" count / progress

### Corrections-not-labels-from-scratch principle

Most users will just want to fix the AI's 2-3 wrong blocks per analysis, not label everything. The default is: all AI blocks are "pending" → user either clicks **Accept** (keeps as-is) or **Edit** (opens the edit dialog). "Mark all clean" button for fully-correct analyses.

## Data model

New Supabase table `pattern_labels`:

```sql
create table public.pattern_labels (
  id uuid primary key default gen_random_uuid(),
  analysis_id uuid references public.video_analyses(id) on delete cascade,
  user_id uuid references auth.users(id) on delete cascade,
  start_time numeric(8,3) not null,
  end_time numeric(8,3) not null,
  name text not null,                  -- pattern family
  variant text,                        -- 'basket', 'with inside turn', null for 'basic', etc.
  count int,                           -- 6 or 8, null for custom
  confidence numeric(3,2),             -- 0.0-1.0 self-reported
  source text not null default 'user', -- 'user' | 'ai_accepted' | 'ai_edited'
  notes text,
  created_at timestamptz default now(),
  updated_at timestamptz default now()
);
-- RLS: user_id = auth.uid() on select/insert/update/delete
-- Index: (analysis_id, start_time) for timeline rendering
```

## Export format

Matches #139's grading harness schema so labels flow directly into evals:

```json
{
  "clip_id": "IMG_9577",
  "dance_start_sec": 8.2,
  "dance_end_sec": 109.6,
  "truth": [
    {"start": 8.2, "end": 11.0, "name": "starter step", "variant": "basic"},
    ...
  ]
}
```

## Scope (v1)

- New `/label` route + label list page
- Label editor with add/edit/delete/accept-AI flow
- `pattern_labels` table + RLS + migration
- Export-all-as-JSON button
- Nav entry in sidebar + mobile nav

## Out of scope (v1)

- Cross-user labeling ("show me labels from judges")
- Inter-rater agreement stats
- Per-pattern leaderboards / gamification

## Follows

- #139 pattern-identification grading harness consumes these labels
- #140 beat-tracker silent-fail (unrelated but both feed the same "make it accurate" goal)

## Related discussion: the ML path

User asked about using a small LLM for pattern identification. Short answer:

1. Small TEXT-only LLMs can't watch video. Not an option.
2. **Best fit right now:** fine-tune Gemini 3 Pro via Vertex AI on labels collected through this feature. No architecture change, just better calibration.
3. **Fast-iteration option:** Claude Sonnet 4.5 with few-shot labeled examples in the prompt (no training; just in-context learning from 5-10 canonical labels).
4. **Small model path (future):** MediaPipe/RTMPose → joint-sequence transformer classifier. Small models work here because feature extraction does the heavy lifting. 2-3 weeks of infra work.

This issue unblocks all four paths — they all need labeled ground truth first.

**Comment by sauravpanda** [comment]
v1 shipped across #144, #146, #152.

- New `/label` route + list page with Export JSON
- Label editor with add / edit / delete / accept-AI flow
- Custom-family + extended counts (10/12/16) for conjoined patterns (#152)
- `pattern_labels` table + RLS + service-role read for evals
- Sidebar nav entry

Out-of-scope items remain valid follow-ups (keyboard shortcuts, mark-all-clean, accepted-AI green state). Open separate issues for those if we want to track them.
